src/vfs2.c

```diff
@@ -1723,13 +1748,18 @@ int vfs2_add_uncommitted(sqlite3_file *file,
 		sums.cksum2 = ByteGetBe32(e->wal_cur_hdr.cksum2);
 	}
 
+	struct vfs2_shm_region0 *r0 = e->shm_regions[0];
+	PRE(e->wal_cursor < REGION0_PGNOS_LEN);
```

Comment:
Yes, forgot to add a TODO here. Creating additional hash tables after the first is tricky because we can't allocate them all at startup like we do currently with region0, but it's necessary for correctness. I'll push a new commit that fixes this.

Reply:
Updates are properly implemented for all hash tables now.